src/HttpCache/EventListener/ConfigureListener.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\HttpCache\EventListener;
+
+use ApiPlatform\Core\Util\RequestAttributesExtractor;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+
+/**
+ * Configures cache HTTP headers for the current response.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+final class ConfigureListener
+{
+    private $maxAge;
+    private $sharedMaxage;
+    private $vary;
+    private $public;
+
+    public function __construct(int $maxAge = null, int $sharedMaxAge = null, array $vary, bool $public = null)
```

Comment:
@jderusse:
> the idea, when using tagged response, is to add a validation header (could be etag or last-modified, don't care). To save bandwidth between users and varnish when cache is hit.

That's incorrect. Varnish already sends 304 response with proper `E-Tag` in that case. There is no need for the backend to send an `E-Tag` or `Last-Modified` header unless the backend intends to return 304 on `vcl_backend_fetch`.